plugins/amd-gpu/fu-amd-gpu-device.c

```diff
@@ -246,9 +246,27 @@ fu_amd_gpu_device_setup(FuDevice *device, GError **error)
 	self->vbios_pn = fu_strsafe((const gchar *)vbios_info.vbios_pn, PART_NUM_STR_SIZE);
 	part = g_strdup_printf("AMD\\%s", self->vbios_pn);
 	fu_device_add_instance_id(device, part);
-	fu_device_set_version_raw(device, vbios_info.version);
-	ver = fu_strsafe((const gchar *)vbios_info.vbios_ver_str, sizeof(vbios_info.vbios_ver_str));
-	fu_device_set_version(device, ver); /* nocheck:set-version */
+
+	tokens =
+	    fu_strsplit((const gchar *)vbios_info.vbios_pn, sizeof(vbios_info.vbios_pn), "-", -1);
+	if (g_strv_length(tokens) >= 3) {
+		guint64 ver;
+
+		if (!fu_strtoull(tokens[2], &ver, 0, G_MAXUINT64, FU_INTEGER_BASE_AUTO, error))
+			return FALSE;
+		fu_device_set_version_format(device, FWUPD_VERSION_FORMAT_NUMBER);
+		fu_device_set_version(device, tokens[2]); /* nocheck:set-version */
+		fu_device_set_version_raw(device, ver);
+	} else {
+		g_autofree gchar *ver = NULL;
+
+		fu_device_set_version_format(device, FWUPD_VERSION_FORMAT_PLAIN);
+		fu_device_set_version_raw(device, vbios_info.version);
+		ver = fu_strsafe((const gchar *)vbios_info.vbios_ver_str,
+				 sizeof(vbios_info.vbios_ver_str));
+		fu_device_set_version(device, ver); /* nocheck:set-version */
+	}
```

Comment:
@hughsie I could be convinced to drop this case (and thus the version string just wouldn't be set) if you would prefer.

Reply:
I think mixing version formats for the same protocol is going to lead us to interesting bugs when we try to compare a `plain` version against a `number` version. I think we either need to stick to `plain` and then format the integer like `%04x` or `%04u` -- or switch to `number` (dropping the compat code) and invalidate the existing firmware on the LVFS.

Reply:
OK I dropped the alternate case.  All the firmware on LVFS is embargo and that's how we caught this problem; I don't see a problem invalidating it (except the newly uploaded one for the test case to pass).

Reply:
> This also might be worth changing the defined protocol for the device if we do change the version format. e.g. com.amd.pspvbflash2

LMK if you think this is critical and I'll respin it, it will obviously be bigger fallout.

Reply:
Ohh if it's all embargo it's fine for me.

Reply:
OK.  I guess on LVFS side you need to set some default that pairs the protocol with `number` instead though?

Reply:
Yup, I can do that when we merge.

Reply:
Done!